contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -470,56 +518,100 @@ contract FeeSharingCollector is
     /**
      * @dev withdraw all of the RBTC balance based on particular checkpoints
      *
-     * RBTC balance consists of:
-     * - rbtc balance
-     * - wrbtc balance which will be unwrapped to rbtc
-     * - iwrbtc balance which will be unwrapped to rbtc
+     * This function will withdraw RBTC balance which is passed as _token param, so it could be either of these:
+     * - rbtc balance or
+     * - wrbtc balance which will be unwrapped to rbtc or
+     * - iwrbtc balance which will be unwrapped to rbtc or
      *
+     * @dev WARNING! This function skips all the checkpoints before '_fromCheckpoint' irreversibly, use with care
+     *
+     * @param _tokens array of either RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT or wrbtc address or iwrbtc address
      * @param _maxCheckpoints  Maximum number of checkpoints to be processed to workaround block gas limit
      * @param _receiver An optional tokens receiver (msg.sender used if 0)
      */
-    function withdrawRBTC(uint32 _maxCheckpoints, address _receiver) external nonReentrant {
-        _withdrawRBTC(_maxCheckpoints, _receiver);
+    function withdrawRbtcToken(
+        address[] calldata _tokens,
+        uint32 _maxCheckpoints,
+        address _receiver
+    ) external nonReentrant {
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            if (_maxCheckpoints <= 0) break;
+            address _token = _tokens[i];
+            uint256 startingCheckpoint = processedCheckpoints[msg.sender][_token];
+
+            (, uint256 endToken) = _withdrawRbtcToken(_tokens[i], _maxCheckpoints, _receiver);
+
+            uint256 _previousUsedCheckpoint = endToken.sub(startingCheckpoint);
+            _maxCheckpoints = uint32(_maxCheckpoints.sub(_previousUsedCheckpoint));
+        }
     }
 
     /**
      * @dev Withdraw all of the RBTC balance based starting from a specific checkpoint
      * The function was designed to skip checkpoints with no fees for users
      *
-     * RBTC balance consists of:
+     * This function will withdraw RBTC balance consists of:
      * - rbtc balance
      * - wrbtc balance which will be unwrapped to rbtc
      * - iwrbtc balance which will be unwrapped to rbtc
      *
      * @dev WARNING! This function skips all the checkpoints before '_fromCheckpoint' irreversibly, use with care
      *
-     * @param _fromCheckpoint Skips all the checkpoints before '_fromCheckpoint'
+     * @param _tokens array of rbtc token to be withdrawn
+     * @param _fromCheckpoints Skips all the checkpoints before array of '_fromCheckpoint'
      *        should be calculated offchain with getNextPositiveUserCheckpoint function
      * @param _maxCheckpoints  Maximum number of checkpoints to be processed to workaround block gas limit
      * @param _receiver An optional tokens receiver (msg.sender used if 0)
      */
-    function withdrawRBTCStartingFromCheckpoint(
+    function withdrawRbtcTokenStartingFromCheckpoint(
```

Comment:
withdrawRbtcToken**s**StartingFromCheckpoint

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/507/commits/695fa69d802d4f5e9417b4a0a8ca9acee1977b98)